libraries/persistent_store/src/buffer.rs

```diff
@@ -0,0 +1,668 @@
+// Copyright 2019-2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{Storage, StorageError, StorageIndex, StorageResult};
+use alloc::borrow::Borrow;
+use alloc::boxed::Box;
+use alloc::vec;
+
+/// Simulates a flash storage using a buffer in memory.
+///
+/// This storage tracks how many times words are written between page erase cycles, how many times
+/// pages are erased, and whether an operation flips bits in the wrong direction (optional).
+/// Operations panic if those conditions are broken. This storage also permits to interrupt
+/// operations for inspection or to corrupt the operation.
```

Comment:
Yes.

Reply:
I was wondering if it's worth mentioning somewhere at the top. So a reader understands this is not the implementation used in the OpenSK device.

Reply:
> So a reader understands this is not the implementation used in the OpenSK device.

Actually another clarification about that point is that this crate is unrelated to OpenSK and Tock (it could be in its own repository and published independently). The storage implementation of OpenSK is going to be in OpenSK code and specific to Tock.

Reply:
But since we use it in OpenSK unit tests, it makes sense here, no?

Reply:
Not really. It would be like mentioning OpenSK in `BTreeMap` documentation because we use them in OpenSK. This is a generic library that has no OpenSK or Tock specificity (this is actually not completely true, there is one optimisation that is not possible due to a design problem in Tock). The library just happens to have been developed as part of the OpenSK project, but it is independent. In particular other embedded projects (like Chromium EC or Tock) could use this library, in which case it would be surprising to have the documentation mention a particular use-case by name.

Reply:
I meant code location. Having it inside OpenSK seems to make sense to me. So I think we agree on this one. Resolved.

Reply:
I'm not sure what you mean :D but if I understand correctly it would become relevant when we reach the PRs about OpenSK, so there's time.